ProcessLib/BoundaryCondition/BoundaryConditionCollection.cpp

```diff
@@ -52,6 +38,14 @@ void BoundaryConditionCollection::addBCsForProcessVariables(
                   std::back_inserter(_boundary_conditions));
     }
 
-    initializeDirichletBCs(_boundary_conditions, _dirichlet_bcs);
+    // for each BC there will be storage for Dirichlet BC, which by default is
+    // empty and has to be filled by the respective BC object if needed.
+    _dirichlet_bcs.resize(_boundary_conditions.size());
```

Comment:
Maybe I misunderstand the comment, but resize sets the _dirichlet_bcs size, so `!_dirichlet_bcs.empty()`